### PR TITLE
Handle only ROS1 SSC in drivers and remove camera related logics

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

Where it removes ros1_camera_drivers as they are not required anymore. This is because live vehicles don't have working camera setup and if we do support it in the future, they will most likely be in ROS2. 
That change triggered the overhaul of drivers_manager which was only meant for ROS1 drivers. 
Since only SSC is in ROS1, we decided to limit the usage of that logic to only ssc and limit extensibility so that future development is forced into ROS2 drivers if we decide to add. 

NOTE: used clang-tidy for the auto format, so please turn on "hide whitespace change" for easier review
Supporting PR: https://github.com/usdot-fhwa-stol/carma-config/pull/387
## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/2079

## Related Jira Key
https://usdot-carma.atlassian.net/browse/CAR-6075

## Motivation and Context
Migration to ROS2 foxy

## How Has This Been Tested?
Local VM tested and verified that the all drivers ready status publishes to /system_alert
image
Following config gets printed from the controller node:
image

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by john-chrosniak — COMMENTED** [review]
This all looks good to me, but I'll let Anish give the final review since he is more familiar with this section of the codebase

File: subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+namespace subsystem_controllers
+{
+SSCDriverManager::SSCDriverManager() {}
+
+SSCDriverManager::SSCDriverManager(const std::string & ssc_driver_name, const long driver_timeout)
+: ssc_driver_name_(ssc_driver_name), driver_timeout_(driver_timeout)
+{
+}
+
+carma_msgs::msg::SystemAlert SSCDriverManager::get_latest_system_alert(
```
[the remark this answers is not in the document]
Comment (MishkaMN) [reply subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp:36]: Yeah that is still the case. The original driver_manager was not removed but it was just renamed to SSCDriverManager . EntryManager was removed completely.

File: subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  }
+}
+
+void SSCDriverManager::update_driver_status(
+  const carma_driver_msgs::msg::DriverStatus::SharedPtr msg, long current_time)
+{
+  // update driver status is only called in response to a message received on driver_discovery.
+  // This topic is only being published in ros1. Check only SSC
+
+  if (ssc_driver_name_.compare(msg->name) == 0) {
```
[the remark this answers is not in the document]
Comment (MishkaMN) [reply subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp]: Done

File: subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      msg->name, current_time);
+
+    latest_ssc_status_entry_ = std::make_shared<Entry>(driver_status);
+  }
+}
+
+bool SSCDriverManager::is_ssc_driver_operational(long current_time)
+{
+  // Manual disable of ssc entry in case ssc wrapper is in ros2
+  if (ssc_driver_name_ == "") {
```
[the remark this answers is not in the document]
Comment (MishkaMN) [reply subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp]: Done

**Comment by MishkaMN** [comment]
Actually decided not to increase the unit test level because all of the uncovered lines come from the node logic which is mainly creating subscriptions and publishing etc. I can include FRIEND_TEST but I don't think it's a good design.

File: subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (adev4a) [reply subsystem_controllers/src/drivers_controller/ssc_driver_manager.cpp:36]: My bad. Nit: can we update the function name to "set"?

File: subsystem_controllers/src/drivers_controller/drivers_controller_node.cpp [hunk 4]
```diff
@@ -19,146 +19,145 @@
...
 
-    if (base_return == CallbackReturn::SUCCESS) {
-      RCLCPP_INFO(get_logger(), "Driver Controller activated!");
-    }
-    else if (base_return == CallbackReturn::FAILURE)
-    {
-      RCLCPP_ERROR(get_logger(), "Driver Controller encountered FAILURE when trying to activate the subsystems... please check which driver failed to activate...");
-    }
+  // Handle fact that parameter vectors cannot be empty
+  if (
```
Comment (adev4a) [inline subsystem_controllers/src/drivers_controller/drivers_controller_node.cpp:63]: It seems like the base subsystem controller does not handle "excluded_namespace_nodes" being empty. Lets add that here as well

File: subsystem_controllers/config/drivers_controller_config.yaml [hunk 5]
```diff
@@ -2,37 +2,34 @@
...
     # Boolean: If this flag is true then all nodes under subsystem_namespace are treated as required in addition to any nodes in required_subsystem_nodes
     full_subsystem_required: false
 
     # Int: The time allocated for system startup in seconds
     startup_duration: 30
 
     # Double: The timeout threshold for essential drivers in ms
-    required_driver_timeout: 3000.0
\ No newline at end of file
+    required_driver_timeout: 3000.0
```
Comment (adev4a) [inline subsystem_controllers/config/drivers_controller_config.yaml:35]: This can be renamed to ssc_driver_timeout for clarity

Comment (MishkaMN) [reply subsystem_controllers/config/drivers_controller_config.yaml]: renamed

**Comment by MishkaMN** [comment]
Docker build in github action is failing at the moment but it is due to msg name change in carma-msgs that is making mock_drivers to fail, but that package is currently being removed in another PR.


The subsystem_controllers package is building okay locally:
image

**Comment by john-chrosniak** [comment]
> Docker build in github action is failing at the moment but it is due to msg name change in carma-msgs that is making mock_drivers to fail, but that package is currently being removed in another PR.

I had fixes for this in #2442 but decided to close it since these packages are currently being removed. If desired, we could merge this PR to fix the build error instead of waiting for them to be removed.

**Review by adev4a — APPROVED** [review]

**Pull request merged**